addons/website_event/models/event_event.py

```diff
@@ -116,28 +141,66 @@ def _compute_is_participating(self):
             registration;
         """
         current_visitor = self.env['website.visitor']._get_visitor_from_request()
-        base_domain = [('event_id', 'in', self.ids), ('state', 'in', ['open', 'done'])]
+        events_in_self = [('event_id', 'in', self.ids)] if self else []
         if self.env.user._is_public() and not current_visitor:
-            events = self.env['event.event']
-        elif self.env.user._is_public():
-            events = self.env['event.registration'].sudo().search(
-                expression.AND([base_domain, [('visitor_id', '=', current_visitor.id)]])
-            ).event_id
+            return self.env['event.event']
+        if self.env.user._is_public():
+            domain = expression.AND([[
+                ('state', '!=', 'cancel'),
+                ('visitor_id', '=', current_visitor.id),
+            ], events_in_self])
```

Comment:
Wouldn't it be easier if this was part of building the "partner_domain"? Seems similar, we want to check the registrations of the person browsing.